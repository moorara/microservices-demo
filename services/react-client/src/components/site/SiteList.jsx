import React from 'react'
import PropTypes from 'prop-types'

import SiteListItem from './SiteListItem'

// Presentational Component
const SiteList = ({ sites }) => (
  <table className="table is-bordered is-striped is-fullwidth is-hoverable">
    <thead>
      <tr>
        <th>Name</th>
        <th>Location</th>
        <th>Priority</th>
        <th>Tags</th>
      </tr>
    </thead>
    <tbody>
      {sites.map(site => <SiteListItem key={site.id} site={site} />)}
    </tbody>
  </table>
)

SiteList.propTypes = {
  sites: PropTypes.array.isRequired
}

export default SiteList
