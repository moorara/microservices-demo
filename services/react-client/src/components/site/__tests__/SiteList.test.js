import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import SiteList from '../SiteList'

describe('SiteList', () => {
  const props = {
    sites: []
  }

  test('renders react component with props', () => {
    props.sites = [
      { id: '1111-1111', name: 'Station 1', location: 'Toronto, ON', priority: 2, tags: [ 'oxygen', 'hydrogen' ] },
      { id: '2222-2222', name: 'Station 2', location: 'Kingston, ON', priority: 3, tags: [ 'gold', 'silver' ] }
    ]
    const wrapper = shallow(<SiteList {...props} />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
    expect(wrapper.find('thead tr th').at(0).text()).toBe('Name')
    expect(wrapper.find('thead tr th').at(1).text()).toBe('Location')
    expect(wrapper.find('thead tr th').at(2).text()).toBe('Priority')
    expect(wrapper.find('thead tr th').at(3).text()).toBe('Tags')
    expect(wrapper.find('SiteListItem').get(0).props.site).toEqual(props.sites[0])
    expect(wrapper.find('SiteListItem').get(1).props.site).toEqual(props.sites[1])
  })
  test('renders react component with props', () => {
    props.sites = [
      { id: '3333-3333', name: 'Station 3', location: 'Ottawa, ON', priority: 4, tags: [ 'helium', 'nitrogen' ] },
      { id: '4444-4444', name: 'Station 4', location: 'Montreal, QC', priority: 5, tags: [ 'carbon', 'iron' ] }
    ]
    const wrapper = shallow(<SiteList {...props} />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
    expect(wrapper.find('thead tr th').at(0).text()).toBe('Name')
    expect(wrapper.find('thead tr th').at(1).text()).toBe('Location')
    expect(wrapper.find('thead tr th').at(2).text()).toBe('Priority')
    expect(wrapper.find('thead tr th').at(3).text()).toBe('Tags')
    expect(wrapper.find('SiteListItem').get(0).props.site).toEqual(props.sites[0])
    expect(wrapper.find('SiteListItem').get(1).props.site).toEqual(props.sites[1])
  })
})
