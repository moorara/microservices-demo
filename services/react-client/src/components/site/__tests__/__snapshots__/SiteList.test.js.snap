// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteList renders react component with props 1`] = `
<table
  className="table is-bordered is-striped is-fullwidth is-hoverable"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Location
      </th>
      <th>
        Priority
      </th>
      <th>
        Tags
      </th>
    </tr>
  </thead>
  <tbody>
    <SiteListItem
      key="1111-1111"
      site={
        Object {
          "id": "1111-1111",
          "location": "Toronto, ON",
          "name": "Station 1",
          "priority": 2,
          "tags": Array [
            "oxygen",
            "hydrogen",
          ],
        }
      }
    />
    <SiteListItem
      key="2222-2222"
      site={
        Object {
          "id": "2222-2222",
          "location": "Kingston, ON",
          "name": "Station 2",
          "priority": 3,
          "tags": Array [
            "gold",
            "silver",
          ],
        }
      }
    />
  </tbody>
</table>
`;

exports[`SiteList renders react component with props 2`] = `
<table
  className="table is-bordered is-striped is-fullwidth is-hoverable"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Location
      </th>
      <th>
        Priority
      </th>
      <th>
        Tags
      </th>
    </tr>
  </thead>
  <tbody>
    <SiteListItem
      key="3333-3333"
      site={
        Object {
          "id": "3333-3333",
          "location": "Ottawa, ON",
          "name": "Station 3",
          "priority": 4,
          "tags": Array [
            "helium",
            "nitrogen",
          ],
        }
      }
    />
    <SiteListItem
      key="4444-4444"
      site={
        Object {
          "id": "4444-4444",
          "location": "Montreal, QC",
          "name": "Station 4",
          "priority": 5,
          "tags": Array [
            "carbon",
            "iron",
          ],
        }
      }
    />
  </tbody>
</table>
`;
